src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -88,6 +96,7 @@ public void setNextReader(AtomicReaderContext reader) {
             } else {
                 childDocs = childDocIdSet.iterator();
             }
+            rootDocs = SearchContext.current().fixedBitSetFilterCache().getFixedBitSetFilter(NonNestedDocsFilter.INSTANCE).getDocIdSet(reader, null);
```

Comment:
Maybe pull the search context from the aggregation context instead? It does not make any difference today but I think it would help if we ever have aggregation unit tests.